lasagne/layers/pool.py

```diff
@@ -406,6 +409,14 @@ def __init__(self, incoming, scale_factor, **kwargs):
             raise ValueError('Scale factor must be >= 1, not {0}'.format(
                 self.scale_factor))
 
+        self.mode = mode
+        if self.mode is None:
+            self.mode = 'repeat'
+
+        if self.mode not in {'repeat', 'bed_of_nails'}:
+            msg = "Mode must be either 'repeat' or 'bed_of_nails', not {0}"
+            raise ValueError(msg.format(self.mode))
```

Comment:
Why should we support `None` here? I wouldn't add this unless required. Minor nitpick: I'd check `mode` before assigning to `self.mode`, that's how most layers do it.
```python
if mode not in ...:
    ...
self.mode = mode
```